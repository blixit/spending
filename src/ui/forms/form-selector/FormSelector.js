import React from 'react';
import styled from 'styled-components';

import Selector from 'ui/inputs/selector/Selector';
import Label from 'ui/forms/form-label/FormLabel';

const ErrorText = styled.div`
  color: red;
`;

const Container = styled.div`
  padding-top: 5px;
  padding-bottom: 5px;
`;

const FormSeletor = ({ label, error, onChange, arabic, values, ...rest }) => {
  const selectorProps = { onChange, arabic, values, ...rest };
  return (
    <Container {...rest} >
      <Label arabic={arabic}>
        <span>{label}</span>
        <Selector {...selectorProps} />
      </Label>
      {error && <ErrorText>{error}</ErrorText>}
    </Container>
  )
};

FormSeletor.displayName = 'FormSeletor';

export default FormSeletor;
