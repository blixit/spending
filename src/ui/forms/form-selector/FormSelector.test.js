import React from 'react';

import render from 'testing/render';

import FormSelector from './FormSelector';

describe('[forms/form-selector] FormSelector', () => {
  const options = [{
    id: 1,
    option: 'item 1'
  }];
  it('renders properly', () => {
    const tree = render(
      <FormSelector
        label='Label'
        onChange={jest.fn()}
        values={options}
      />
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
