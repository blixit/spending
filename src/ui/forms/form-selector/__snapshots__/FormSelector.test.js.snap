// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[forms/form-selector] FormSelector renders properly 1`] = `
<div
  className="sc-bxivhb kKnJWp"
>
  <label
    className="sc-bwzfXH cJnpOh"
  >
    <span>
      Label
    </span>
    <select
      className="sc-bdVaJa kNYkjY"
      onChange={[MockFunction]}
    >
      <option>
        item 1
      </option>
    </select>
  </label>
</div>
`;
